usecases/auth/authorization/controller.go

```diff
@@ -77,18 +70,52 @@ func (m *AuthzController) GetRole(name string) (*models.Role, error) {
 	return roles[0], nil
 }
 
-func (m *AuthzController) DeleteRole(name string) error {
-	policies, err := m.enforcer.GetPolicies(&name)
+func (c *AuthzController) DeleteRole(name string) error {
+	policies, err := c.rbac.GetPolicies(&name)
 	if err != nil {
 		return err
 	}
-	if err := m.enforcer.RemovePolicies(policies); err != nil {
+	if err := c.rbac.RemovePolicies(policies); err != nil {
 		return err
 	}
 	return nil
 }
 
-func (m *AuthzController) rolesFromPolicies(policies []*rbac.Policy) ([]*models.Role, error) {
+func (c *AuthzController) AddRolesForUser(user string, roles []string) error {
+	return c.rbac.AddRolesForUser(user, roles)
+}
+
+func (c *AuthzController) GetRolesForUser(user string) ([]string, error) {
+	return c.rbac.GetRolesForUser(user)
+}
+
+func (c *AuthzController) GetUsersForRole(role string) ([]string, error) {
+	return c.rbac.GetUsersForRole(role)
+}
+
+func (c *AuthzController) DeleteRolesForUser(user string, roles []string) error {
```

Comment:
could we [embed](https://gobyexample.com/struct-embedding) the rbac struct, so we don't need all of these wrapper functions? (not sure if I ever tried it, so might not work)

Reply:
That's the way the `AuthzController` struct is created but adding these wrappers ensures that the packages higher up don't depend on `rbac.Manager`, which is quite low down in the encapsulation. But the change to the tests below may alter this

Reply:
what I mean is

```
type AuthzController struct {
   rbacManager
}
```

instead of

```
type AuthzController struct {
	rbac rbacManager
}
```

then all these wrappers could go away as they are just duplicating the signatures of the rbacManager (or if the interface is not needed anymore, the implementation)

But just an idea, never actually used it

Reply:
Ahh okay, I'll give it a shot!